apis/python/tests/test_experiment_query.py

```diff
@@ -0,0 +1,591 @@
+import sys
+from typing import List, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pytest
+from scipy import sparse
+
+import tiledbsoma as soma
+from tiledbsoma.experiment_query import AxisQuery, ExperimentAxisQuery, X_as_series
+
+"""
+WIP tracker - delete when complete.  Missing tests:
+* query by coords when there is >1 dimension
+* error check handling of Dense X/obsm/varm
+"""
+
+
+@pytest.fixture(scope="function")
+def obs(tmp_path, n_obs) -> soma.DataFrame:
+    return make_dataframe((tmp_path / "obs").as_posix(), n_obs)
+
+
+@pytest.fixture(scope="function")
+def var(tmp_path, n_vars) -> soma.DataFrame:
+    return make_dataframe((tmp_path / "var").as_posix(), n_vars)
+
+
+@pytest.fixture
+def X_layer_names():
+    return ["raw"]
+
+
+@pytest.fixture
+def obsp_layer_names():
+    return None
+
+
+@pytest.fixture
+def varp_layer_names():
+    return None
+
+
+@pytest.fixture(scope="function")
+def soma_experiment(
+    tmp_path,
+    n_obs,
+    n_vars,
+    obs,
+    var,
+    X_layer_names,
+    obsp_layer_names,
+    varp_layer_names,
+):
+    return make_experiment(
+        tmp_path,
+        n_obs,
+        n_vars,
+        obs,
+        var,
+        X_layer_names,
+        obsp_layer_names,
+        varp_layer_names,
+    )
+
+
+@pytest.mark.xfail(
+    # This test fails on Python 3.10+ due to a bug in typeguard. The bug
+    # is tripped due to use of a TypedDict. Remove
+    # work-around when the typeguard issue is fixed AND released.
+    # Underlying issue:
+    #   https://github.com/agronholm/typeguard/issues/242
+    # Related to _when_ we might see a release:
+    #   https://github.com/agronholm/typeguard/issues/257
+    sys.version_info.major == 3 and sys.version_info.minor >= 10,
```

Comment:
`sys.version_info >= (3, 10)`

Reply:
:man_facepalming: someday i need to learn Python...